src/dashboard/src/components/advanced_search.py

```diff
@@ -179,7 +179,42 @@ def _normalize_date(date):
     except ValueError:
         raise ValueError("Invalid date received ({}); ignoring date query".format(date))
 
-def query_clause(index, queries, ops, fields, types):
+def filter_search_fields(search_fields, index=None, doc_type=None):
+    """
+    Given search fields which search nested documents with wildcards (such as "transferMetadata.*"), returns a list of subfields filtered to contain only string-type fields.
+
+    When searching all fields of nested documents of mixed types using query_string queries, query_string queries may fail because the way the query string is interpreted depends on the type of the field being searched.
+    For example, given a nested document containing a string field and a date field, a query_string of "foo" would fail when Elasticsearch attempts to parse it as a date to match it against the date field.
+    This function uses the actual current mapping, so it supports automatically-mapped fields.
+
+    Sample input and output, given a nested document containing three fields, "Bagging-Date" (date), "Bag-Name" (string), and "Bag-Type" (string):
+    ["transferMetadata.*"] #=> ["transferMetadata.Bag-Name", "transferMetadata.Bag-Type"]
+    """
```

Comment:
Sphinx-style docs of params?  Specifically, is search_fields a list or something else? Code implies it's a list, but nice to know explicitly it's not supposed to be a str.